test/project_types/extension/extension_test_helpers.rb

```diff
@@ -18,5 +18,51 @@ module Stubs
       autoload :ArgoScript, "project_types/extension/extension_test_helpers/stubs/argo_script"
       autoload :FetchSpecifications, "project_types/extension/extension_test_helpers/stubs/fetch_specifications"
     end
+
+    def self.test_specification_handler
+      specification_handlers = DummySpecifications.build(
+        custom_handler_root: File.expand_path("../", __FILE__),
+        custom_handler_namespace: ::Extension::ExtensionTestHelpers,
+      )
+
+      specification_handler = specification_handlers["TEST_EXTENSION"]
+      if specification_handler.nil?
+        raise "Unable to retrieve specification handler due to broken test setup"
+      end
+
+      specification_handler
+    end
+
+    def self.fake_extension_project(with_mocks:,
+      api_key: "TEST_KEY",
+      api_secret: "TEST_SECRET",
+      title: "Test",
+      type_identifier: test_specification_handler.identifier,
+      registration_id: 55)
+
+      @context = TestHelpers::FakeContext.new(root: "/fake/root")
+
+      @project = FakeExtensionProject.new(
```

Comment:
Please `@project` with a local variable `project` to avoid leaking global state.